Feature 5455 rewrite the add header script

joeldierkes opened the pull request:
<!-- Please read https://rucio.cern.ch/documentation/contributing before submitting a pull request -->

mlassnig commented on tools/add_header:
```diff
@@ -283,6 +283,75 @@ class PythonHeaderTemplate(HeaderTemplate):
+        file_creation_year = get_git_file_creation_year(self.file_path)
+
+        shebag_string = BashHeaderTemplate._get_file_shebag(self.file_path) or ""
+        copyright_string = "# Copyright CERN" \
```
I would still suggest to use the full name like this:

European Organization for Nuclear Research (CERN)

instead of just the abbreviation

CERN


Also, file_creation_year will be wrong on some of them due to the way some of the files were moved around in git history rewrites. Wouldn't it be easier just to use the overall project duration and simply say "since 2012" everywhere?

joeldierkes replied:
1. Yeah we can change that. I'd create a separate PR for this, since we also need to change this in the python files.
2. https://github.com/rucio/rucio/blob/47e93cfbe5887071c70de4ba815c1bbdddfac2ce/tools/add_header#L60 The `--follow` option in git should resolve this. This is why the year changes on some files. However we can set it fixed to 2012.

joeldierkes committed:
Documentation: Change the python header string #5455

The python header contains the creation year of the file. While git does a good
job of tracking file movements and history, it makes more sense to use the
projects creation here.

"CERN" is a bit to short as a name. We will use the full name here.

joeldierkes committed:
Documentation: Add a bash header template #5455

The bash files should have sanitized headers as well.

joeldierkes committed:
Documentation: Fix file headers #5455
